docs/proposals/perf.md

```diff
@@ -0,0 +1,166 @@
+
+### Test Framework
+<img src="../images/perf/perf-test-framework.png">
+
+### Test Tools
+* [Kubemark](https://github.com/kubernetes/kubernetes/tree/master/test/kubemark)
+* [Fortio](https://github.com/fortio/fortio)
+
+### Metrics Tools
+* [Prometheus](https://github.com/prometheus/prometheus)
+* [Grafana](https://github.com/grafana/grafana)
+
+## Test Scenarios
+
+### 1. Edge Nodes join in K8S Cluster
+<img src="../images/perf/perf-edgenodes-join-cluster.png">
+
+Test Cases:
+* Measure Edge Nodes join in K8S Cluster startup time.
+
+  Different numbers of Edge Nodes need be tested.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+  This test case ends with all Edge Nodes are in `Ready` status.
+
+### 2. Create Device/Device Model from Cloud
+<img src="../images/perf/perf-create-device.png">
+
+This scenario is expected to measure the northbound API of KubeEdge.
+
+Test Cases:
+* Measure the latency between K8S API Server and KubeEdge Cloud Part.
+* Measure the throughput between K8S API Server and KubeEdge Cloud Part.
+
+### 3. Report Device Status to Edge
+<img src="../images/perf/perf-report-devicestatus.png">
+
+This scenario is expected to measure the southbound API of KubeEdge.
+
+Test Cases:
+* Measure the latency between KubeEdge Edge Part and device.
+
+  Device numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+* Measure the throughput between KubeEdge Edge Part and device.
+
+  Device numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+As the result of the latency and throughput with different device numbers,
+we can evaluate scalability of devices for KubeEdge Edge Part.
+Measure the capacity of devices can be supported by KubeEdge Edge Part.
+<img src="../images/perf/perf-multi-devices.png">
+
+Different protocols are considered to test between KubeEdge Edge Part and devices.
+E.g. Bluetooth, MQTT, ZigBee, BACnet and Modbus and so on.
+Currenly less than 20ms latency can be accepted in Edge IoT scenario.
+Two kinds of test cases can be adopted: emulators of different devices and actual devices.
+
+### 4. Application Deployment from Cloud to Edge
+<img src="../images/perf/perf-app-deploy.png">
+
+This scenario is expected to measure the performance of KubeEdge from Cloud to Edge.
+
+Test Cases:
+* Measure the pod startup time with docker images in the Edge Node.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20]`.
+
+* Measure the pod startup time without docker images in the Edge Node.
+
+  Edge Nodes numbers are one of `[1, 10, 20, 50, 100, 200]`.
+
+  Pods numbers per Edge Node are one of `[1, 2, 5, 10, 20]`.
+
+As the result of the pod startup time, we can evaluate scalability of KubeEdge Edge Nodes.
+Measure the capacity of Edge Nodes can be supported by KubeEdge Cloud Part.
+<img src="../images/perf/perf-multi-edgenodes.png">
+
+The simulation of generating virtual KubeEdge Edge Nodes are being analyzed with `Kubemark` framework.
```

Comment:
we need to have a design on how are we planning to simulate edge node . will it be covered in a separate PR?

Reply:
@edisonxiang , can you please push the different deployment scenario's added in design doc, where i have explained the edge node simulations under the deployments.

Reply:
Echo what @rohitsardesai83  called out here. We are using kubemark to test control plane. Need to evaluate whether kubemark's node simulation works or not. 

Besides, we need to evaluate the edge node local performance. After edgehub receives metadata,  how long it takes for edged to take effect. Also how many pods can we run on edge node provided different level constraint of resources (1CPU, 256MB； 2CPU 1GB, etc.)

Reply:
+1

Reply:
Thanks, we will update later

Reply:
Thanks @CindyXing. We have done the research for kubemark.
We found it is not suitable for us to do perf test in kubeedge.
We will not use kubemark. 
Please find more from this link.
https://docs.google.com/presentation/d/1oVB92dLNKMMVJOzfvsdrRthpPwAJxSDlrwe0dkG_R4o/edit#slide=id.g563407762e_0_0

Reply:
That makes sense. Thanks @edisonxiang. One thing, can we run the hollowedgenode at a separate vm? This way we could launch more edge nodes and not impact the k8s master. I might misunderstand the slide 3 chart.